src/main/java/edu/kis/powp/jobs2d/drivers/DriverMonitoringDecorator.java

```diff
@@ -0,0 +1,44 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import java.util.logging.Logger;
+
+public class DriverMonitoringDecorator implements Job2dDriver {
+    protected final Job2dDriver driver;
+    private int headDistance = 0;
+    private int opDistance = 0;
+    private final Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+    private int currentX = 0;
+    private int currentY = 0;
+
+
+    public DriverMonitoringDecorator(Job2dDriver driver) {
+        this.driver = driver;
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        int distance = calculateDistance(currentX, currentY, x, y);
+        headDistance += distance;
+        logger.info("Set position to (" + x + ", " + y + ")\n   head distance = " + headDistance + "\n   op. distance = " + opDistance);
```

Comment:
This (logging) should be on option rather then the only way to see distances. The solution is not OCP, e.g., if I would like to see the distance as a bar in GUI.